### PR TITLE
Update gitignore and setup.py

### LINKED ISSUE #23: Coverage of .gitignore is incomplete
The top-level `build` and `networkx-metis.egg-info` directories are missed. GitHub has a template: https://github.com/github/gitignore/blob/master/Python.gitignore. We can take a part of that.

### PR DESCRIPTION
**Pull request by OrkoHunter**
Fixes #23 and removes `>=2.0` dependency.

### CODE REVIEW

**Comment by OrkoHunter** [comment]
Before we merge this, there's a failure on travis
```
 File "/home/travis/build/OrkoHunter/networkx-metis/networkx/addons/metis/enums.py", line 113, in MetisDbgLvl
    contiginfo, memory) = [1 << x for x in range(9) + [11]]
nose.proxy.TypeError: unsupported operand type(s) for +: 'range' and 'list'
```
We need to fix that.

**Comment by OrkoHunter** [comment]
Done

File: networkx/addons/metis/enums.py [hunk 1]
```diff
@@ -110,7 +110,7 @@ class MetisDbgLvl(enum.IntEnum):
 
     default = -1
     (info, time, coarsen, refine, ipart, moveinfo, sepinfo, conninfo,
-     contiginfo, memory) = [1 << x for x in range(9) + [11]]
+     contiginfo, memory) = [list(1 << x for x in range(9)) + [11]]
```
Comment (ysitu) [inline networkx/addons/metis/enums.py]: Remove the outmost brackets.

**Comment by OrkoHunter** [comment]
We can merge this PR after everything goes well on travis.

File: networkx/addons/metis/enums.py [hunk 2]
```diff
@@ -110,7 +110,7 @@ class MetisDbgLvl(enum.IntEnum):
 
     default = -1
     (info, time, coarsen, refine, ipart, moveinfo, sepinfo, conninfo,
-     contiginfo, memory) = [1 << x for x in range(9) + [11]]
+     contiginfo, memory) = list(1 << x for x in range(9)) + [11]
```
Comment (ysitu) [inline networkx/addons/metis/enums.py]: Oh, sorry. I misread the code. It should be

```python
[1 << x for x in list(range(9)) + [11]]
```

**Comment by OrkoHunter** [comment]
Well, [this line](https://github.com/networkx/networkx-metis/blob/master/networkx/addons/metis/types.py#L3) needs fix too over Python 3 and pypy 3.

**Comment by OrkoHunter** [comment]
[Build](https://travis-ci.org/OrkoHunter/networkx-metis) is smooth now. We can merge this and #16 after I update that.

**Pull request merged**